app/lib/clients/bambooHR_client.rb

```diff
@@ -0,0 +1,45 @@
+require 'bamboozled'
+require 'date'
+
+
+class Clients::BambooHRClient
+    client = Bamboozled.client(subdomain: AppEnv['BAMBOOHR_SUBDOMAIN'], api_key: AppEnv['BAMBOOHR_API_KEY'])
+    # birthday = client.employee.all("birthday")
+    # anniversary = client.employee.all("hireDate")
+    day_before_yesterday = (Date.today - 2).to_formatted_s(:iso8601)
+    yesterday = (Date.today - 1).to_formatted_s(:iso8601)
+    today = (Date.today).to_formatted_s(:iso8601)
+    tomorrow = (Date.today + 1).to_formatted_s(:iso8601)
+    day_afer_tomorrow = (Date.today + 2).to_formatted_s(:iso8601)
+    time_off = client.time_off.whos_out(Time.now, today)
+
+    # puts birthday
+    # puts anniversary
+    puts "time off: "+time_off.to_s
+
+
+    employees_info = client.report.custom(["id", "birthday", "hireDate", "isPhotoUploaded"], format = "JSON")
+    employees = []
+
+    for employee in employees_info
+        #and employee['birthday'] != "" included because 2 of the employees in the test company do not have birthdays listed
+        if day_before_yesterday.include? employee['birthday'] or yesterday.include? employee['birthday'] or today.include? employee['birthday'] or tomorrow.include? employee['birthday'] or day_afer_tomorrow.include? employee['birthday'] and employee['birthday'] != ""
+            puts "birthday: "+employee['birthday'] 
+            employees.append(employee)
+        end
+        #employee['hireDate'].include? '8-13' was included because no employees were hired during the current dates so I added a date where I knew employees were hired on to test that this was working correctly
+        if day_before_yesterday.include? employee['hireDate'] or yesterday.include? employee['hireDate'] or today.include? employee['hireDate'] or tomorrow.include? employee['hireDate'] or day_afer_tomorrow.include? employee['hireDate'] or employee['hireDate'].include? '8-13' 
+            puts "hireDate: "+employee['hireDate'] 
+            employees.append(employee)
+        end
+        time_off.each{ |i| if i['employeeId'].to_s.include? employee['id']
+            puts "employees out: "+employee['id']
+            employees.append(employee)
```

Comment:
iterating/appending could also be done with map/compact:
https://ruby-doc.org/core-2.5.7/Enumerable.html#method-i-map
https://ruby-doc.org/core-2.5.7/Array.html#method-i-compact